### PR TITLE
#263 | Address input

### LINKED ISSUE #263: Implement address input
## Overview
We need a reusable address input to improve UX/consistency and reduce code duplication

## Acceptance criteria
- Address input can be taken from teloscan and styled to match web wallet
- address input on send page should be replaced with new component
- Validation for Receiving Address input field:
    - Valid checksum accepted
    - Invalid checksum rejected
    - All lowercase accepted but warning shown:
    > Unchecksummed EVM address detected. Proceed with caution and verify the source and address before confirming the transaction.

### PR DESCRIPTION
**Pull request by EJSG**
# Fixes #263 

## Description
This PR adds an address input and adds it to the send page

note: due to a limitation of the quasar input, it is not possible to have the size of the input be reactive between desktop and mobile. i tried every method in the book, including re-rendering the input several times. it just dont work. however on initial load it works fine; on mobile youll see a text area to show the whole input. on desktop youll see a normal sized input

## Test scenarios
**Demo page**
- go to https://deploy-preview-353--wallet-staging.netlify.app/
- log in
- go to https://deploy-preview-353--wallet-staging.netlify.app/demos/inputs
- play around with the address input
    - it should show a warning state when you enter a lowercase valid address
    - it should show an error state when you enter an invalid addresss
    - it should show an error state when you enter an address with invalid checksum (e.g. paste your own address and replace one uppercase letter with a lower one)
    - it should be disabled when appropriate
    - it should show a green success state when enabled, if there is no warning or error

**Send page**
- go to https://deploy-preview-353--wallet-staging.netlify.app/evm/send
- enter a valid amount to send
- enter an invalid address (invalid checksum or not an address at all)
    - send button should be disabled
- enter lowercased address (chrome devtools console > copy your address from metamask > execute `'(address here)'.toLowerCase()`
    - send button should be enabled
    - warning state should be shown
- enter normal address
    - send button should be enabled
    - address input should have a green success state
- send yourself a token
    - metamask should have the correct receipt address

## Checklist:

-   [x] I have performed a self-review of my own code
-   [x] I have commented my code, particularly in hard-to-understand areas
-   [x] I have cleaned up the code in the areas my change touches
-   [x] My changes generate no new warnings
-   [x] I have checked my code and corrected any misspellings
-   [x] I have removed any unnecessary console messages
-   [x] I have included all english text to the translation file

### CODE REVIEW

File: src/boot/i18n.js [hunk 1]
```diff
@@ -5,6 +5,7 @@ import messages from 'src/i18n';
 export default boot(({ app }) => {
     const i18n = createI18n({
         locale: 'en-us',
+        legacy: false,
```
Comment (EJSG) [inline src/boot/i18n.js:8]: fix error when using i18n in `script setup`

File: src/css/global/_typography.scss [hunk 2]
```diff
@@ -133,30 +133,110 @@ body {
...
+    }
+
+    &--warning {
+        @include warning-text;
+    }
+
+    &--error {
+        @include error-text;
+    }
+}
```
Comment (EJSG) [inline src/css/global/_typography.scss:222]: updating util classes to have `!important` and use mixins

File: src/components/evm/inputs/CurrencyInput.vue [hunk 3]
```diff
@@ -1097,10 +1099,10 @@ export default defineComponent({
 
         position: absolute;
         top: -24px;
-        left: 0;
         right: 0;
-        text-align: right;
+        width: max-content;
         cursor: pointer;
+        padding: 0 0 8px 8px; // make clickable area slightly larger
```
Comment (EJSG) [inline src/components/evm/inputs/CurrencyInput.vue:1105]: fixing an issue where the max amount container is clickable all the way to the left of the input. now only the text is clickable

**Review by pmjanus — APPROVED** [review]

**Review by Viterbo — APPROVED** [review]
Good Job!

**Pull request merged**